packages/lit-html/src/parts.ts

```diff
@@ -78,15 +138,52 @@ export const setPartValue = <T extends Part>(
     }
     const newValues = [...(part._value as Array<unknown>)];
     newValues[index] = value;
-    (part as AttributePart)._setValue(newValues, 0);
+    (part as AttributePart)._setValue(newValues, directiveParent, 0);
   } else {
-    part._setValue(value);
+    part._setValue(value, directiveParent);
   }
   return part;
 };
 
+/**
+ * Sets the value of a directive asynchronously, outside the normal
```

Comment:
When I read this I wonder "what's a directive value?" Does a directive have a single value? I know this means the value typically returned by the directive's `update()` method, but the value phrasing reads odd to me. I wonder if we should talk about the directive "result" instead?

Reply:
Maybe the API should be `updateDirectiveResult`?

Reply:
In https://github.com/Polymer/lit-html/pull/1432 I'm moving this to an instance method on `DisconnectableDirective`, so maybe just `setPartValue` there would be good?